libraries/AP_XRCE_Client/AP_XRCE_Client.cpp

```diff
@@ -0,0 +1,118 @@
+#include "AP_XRCE_Client.h"
+
+AP_HAL::Semaphore* AP_XRCE_Client::get_clientsemaphore()
+{
+    return &(AP::ahrs().get_semaphore());
+}
+
+bool AP_XRCE_Client::init()
+{
+    
+    _csem = AP_XRCE_Client::get_clientsemaphore();
+    if (!_csem){
+        hal.console->printf("No semaphore");
+        return false;
+    }
+ 
+    if (this->use_serial) {
+        this->fd=open("/dev/ttyUSB1", O_RDWR | O_NOCTTY | O_NONBLOCK);
```

Comment:
You should get a serial object from the hal that way you ensure that the port is configured correctly

Reply:
Also this should be made available on other serial ports as well... maybe check how other libraries are implemented and the parameter system that sets them up.
https://github.com/ArduPilot/ardupilot/blob/3386cbb0dfece31d0acffdcb2147abd7df2643c6/libraries/AP_SerialManager/AP_SerialManager.cpp#L412